tests/unit/test_utils.py

```diff
@@ -46,6 +46,18 @@ def test_custom_username_field(self):
         get_id = get_identity(Context({}), user=MyUser(identity='fake_id'))
         assert get_id == 'fake_id'
 
+    def test_custom_identity(self):
+        get_id = get_identity(Context({
+            'foo_provider_identity': 'bar',
+            'analytical_identity': 'baz',
+        }), prefix='foo_provider')
+        assert get_id == 'bar'
+
+        get_id = get_identity(Context({
+            'analytical_identity': 'baz',
+        }), prefix='foo_provider')
+        assert get_id == 'baz'
```

Comment:
Could you split this up into two separate tests?

Reply:
Done.